lib/service/connect.go

```diff
@@ -1041,16 +1047,39 @@ func (process *TeleportProcess) rotate(conn *Connector, localState auth.StateV2,
 	}
 }
 
-// newClient attempts to connect directly to the Auth Server. If it fails, it
-// falls back to trying to connect to the Auth Server through the proxy.
-// The proxy address might be configured in process environment as apidefaults.TunnelPublicAddrEnvar
-// in which case, no attempt at discovering the reverse tunnel address is made.
-func (process *TeleportProcess) newClient(authServers []utils.NetAddr, identity *auth.Identity) (*auth.Client, error) {
+// newClient attempts to tunnel to either the proxy server or auth server
+// When a proxy server is specified (config v3 onwards), it will only attempt to tunnel to the proxy
+// and will error if the connection fails.
+// When only auth servers are specified (config v1 and v2 can have multiple auth servers, v3 only allows one)
+// It will attempt to direct dial the auth server, and fallback to trying to tunnel connect to the
```

Comment:
Does this mean the fallback to tunneling happens even for config v3 when you specify a single auth_server?

Reply:
I didn't originally have the config version in `newClient`, so it would've falled back for a single `auth_server` with config v3.

I've changed this now, config v3 will attempt to only either tunnel to the proxy server (if configured) or direct dial to the auth server (if configured), with no fallbacks.

Config v1 and v2 keep the same functionality as before.